modules/pipeline/providers/engine/provider.go

```diff
@@ -63,41 +62,20 @@ func (p *provider) Init(ctx servicehub.Context) error {
 	// dbclient
 	p.dbClient = &dbclient.Client{Engine: p.MySQL.DB()}
 
-	// cluster info
-	// TODO setup inside clusterinfo provider later
-	p.initClusterInfoUntilSuccess(ctx)
-
 	// action executor manager
 	_, cfgChan, err := p.dbClient.ListPipelineConfigsOfActionExecutor()
 	if err != nil {
 		return err
 	}
 	mgr := actionexecutor.GetManager()
 	p.actionExecutorMgr = mgr
-	if err := mgr.Initialize(ctx, cfgChan); err != nil {
+	if err := mgr.Initialize(ctx, cfgChan, p.ClusterInfo); err != nil {
```

Comment:
I think action executor manager should be provider later :)

Reply:
maybe should action executor should be provider
could create task for it